vscode/webviews/promptEditor/plugins/atMentions/atMentions.tsx

```diff
@@ -115,23 +114,26 @@ export default function MentionsPlugin(): JSX.Element | null {
                 })
                 .slice(0, SUGGESTION_LIST_LENGTH_LIMIT) ?? []
         )
-    }, [results])
+    }, [results, model, tokenAdded])
 
     // biome-ignore lint/correctness/useExhaustiveDependencies: Intent is to update whenever `options` changes.
     useEffect(() => {
         update()
     }, [options, update])
 
-    // Listen for changes to ContextItemMentionNode to update the token count.
-    // This updates the token count when a mention is added or removed.
-    editor.registerMutationListener(ContextItemMentionNode, node => {
-        const items = toSerializedPromptEditorValue(editor)?.contextItems
-        if (!items?.length) {
-            setTokenAdded(0)
-            return
-        }
-        setTokenAdded(items?.reduce((acc, item) => acc + (item.size ? item.size : 0), 0) ?? 0)
-    })
+    useEffect(() => {
+        // Listen for changes to ContextItemMentionNode to update the token count.
+        // This updates the token count when a mention is added or removed.
+        const unregister = editor.registerMutationListener(ContextItemMentionNode, node => {
+            const items = toSerializedPromptEditorValue(editor)?.contextItems
+            if (!items?.length) {
+                setTokenAdded(0)
+                return
+            }
+            setTokenAdded(items?.reduce((acc, item) => acc + (item.size ? item.size : 0), 0) ?? 0)
+        })
+        return unregister
+    }, [editor])
```

Comment:
```suggestion
    }, [])
```

This effect will get re-rendered whenever users add a new @-mention (which changes the editor state) if `editor` is listed as a dependency. If we only want to register it once, does removing `editor` work?

Reply:
It seems to work. Is it true that `editor` changes when the editor state changes? I would have thought the `editor` var remains the same, but the object value might change deeply.

Reply:
> Is it true that editor changes when the editor state changes?

I added a console log within the useEffect, and it was called when the editor state changes, but I can't seem to reproduce it on the latest main, so I must have used the wrong branch when I was testing this... Sorry for the false alarm.